air/src/proof/context.rs

```diff
@@ -93,6 +93,45 @@ impl Context {
     }
 }
 
+impl<E: StarkField> ToElements<E> for Context {
+    /// Converts this [Context] into a vector of field elements.
+    ///
+    /// The elements are layed out as follows:
+    /// - trace layout info [1 or more elements].
+    /// - field modulus bytes [2 field elements].
+    /// - field extension and FRI parameters [1 element].
+    /// - grinding factor [1 element].
+    /// - blowup factor [1 element].
+    /// - number of queries [1 element].
+    /// - trace length [1 element].
+    /// - trace metadata [0 or more elements].
+    fn to_elements(&self) -> Vec<E> {
+        // convert trace layout
+        let mut result = self.trace_layout.to_elements();
+
+        // convert field modulus bytes into 2 elements
+        let num_modulus_bytes = self.field_modulus_bytes.len();
+        let (m1, m2) = self.field_modulus_bytes.split_at(num_modulus_bytes / 2);
+        result.push(bytes_to_element(m1));
+        result.push(bytes_to_element(m2));
+
+        // convert proof options and trace length to elements
+        result.append(&mut self.options.to_elements());
+        result.push(E::from(self.trace_length as u64));
+
+        // convert trace metadata to elements; this is done by breaking trace metadata into chunks
+        // of bytes which are slightly smaller than field element requires, and then converting
```

Comment:
nit: sentence containing "requires" should probably be reformulated

Reply:
This sentence from the same PR should work: "..is slightly smaller than the number of bytes needed to encode a field element"

Reply:
Thank you! Fixed!